### PR TITLE
Add the grade circle to the rubric

### PR DESCRIPTION
**Pull request by shichongrui**
refs: MBL-12301, MBL-12302
affects: Student
release note: none

### CODE REVIEW

File: Student/Student/Submissions/SubmissionRubric/RubricViewController.swift [hunk 1]
```diff
@@ -52,6 +55,31 @@ class RubricViewController: UIViewController {
 }
 
 extension RubricViewController: UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, referenceSizeForHeaderInSection section: Int) -> CGSize {
+        if presenter.assignments.first?.useRubricForGrading ?? false {
+            return CGSize(width: collectionView.frame.width, height: 156)
+        } else {
+            return CGSize(width: collectionView.frame.width, height: 0)
```
Comment (bluwave) [inline Student/Student/Submissions/SubmissionRubric/RubricViewController.swift]: `CGSize.zero`?

File: Core/Core/Views/GradeCircleView.swift [hunk 2]
```diff
@@ -16,6 +16,29 @@
...
+
+    public override init(frame: CGRect) {
+        gradeCircleView = GradeCircleView(frame: CGRect(x: 0, y: topPadding, width: frame.width, height: frame.height))
+        super.init(frame: frame)
+        guard let gradeCircleView = gradeCircleView else {
+            return
+        }
+        addSubview(gradeCircleView)
+
+        let border = UIView(frame: CGRect(x: 16, y: frame.height, width: frame.width - 32, height: 1))
```
Comment (bluwave) [inline Core/Core/Views/GradeCircleView.swift]: do we want the height to be `1.0 / UIScreen.main.scale`?  Not a necessity.  Sometimes it makes borders just a hair smaller.

File: Core/Core/Views/GradeCircleView.swift [hunk 3]
```diff
@@ -16,6 +16,29 @@
 
 import UIKit
 
+let topPadding: CGFloat = 12
+public class GradeCircleReusableView: UICollectionReusableView {
+//    @IBOutlet public weak var gradeCircleView: GradeCircleView?
```
Comment (bluwave) [inline Core/Core/Views/GradeCircleView.swift]: remove?

File: Core/Core/Views/GradeCircleView.swift [hunk 4]
```diff
@@ -16,6 +16,29 @@
 
 import UIKit
 
+let topPadding: CGFloat = 12
```
Comment (bluwave) [inline Core/Core/Views/GradeCircleView.swift]: is this used anywhere besides this view?  May be better to stick it in class.

File: Core/Core/Views/GradeCircleView.swift (hunk 2 above)
Comment (vanwagonet) [reply Core/Core/Views/GradeCircleView.swift]: Dave has had me change a few of those hairlines to be 1 pt. I think we mostly don't want hairlines in our current design.

Comment (bluwave) [reply Core/Core/Views/GradeCircleView.swift]: oh, good to know. Scratch that then.

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-12301](https://instructure.atlassian.net/browse/MBL-12301)
[MBL-12302](https://instructure.atlassian.net/browse/MBL-12302)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 93.53% | 93.53% | 0%
**Student** | 92.27% | 92.27% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.65% | 92.65% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Student/Student/Submissions/SubmissionRubric/RubricViewController.swift [hunk 5]
```diff
@@ -52,6 +55,31 @@ class RubricViewController: UIViewController {
...
+        }
+    }
+
+    func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
+        switch kind {
+        case UICollectionView.elementKindSectionHeader:
+            guard
+                let gradeView = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: "\(GradeCircleReusableView.self)", for: indexPath) as? GradeCircleReusableView,
```
Comment (vanwagonet) [inline Student/Student/Submissions/SubmissionRubric/RubricViewController.swift]: we have a Core extension for this: `collectionView.dequeue(GradeCircleReusableView.self, ofKind: kind, for: indexPath)`

That extension could also be improved to look more like the table view one. `let gradeView: GradeCircleReusableView  = collectionView.dequeue(ofKind: kind, for: indexPath)`

**Pull request merged**